docker-compose.yml

```diff
@@ -1866,7 +1801,6 @@ services:
         jdk: 17
         maven: ${MAVEN}
         node: ${NODE}
-        go: ${GO}
```

Comment:
Can we always use the latest Go in `ci/docker/conda-integration.dockerfile`?

https://github.com/apache/arrow-go/blob/12482ca3eb1cd4da21c45325e174f1dac871281b/dev/release/verify_rc.sh#L137-L156 may help us.

Reply:
I've tried something here: https://github.com/apache/arrow/pull/44143/commits/3d2516dec16dccf542cdc610a5ddf2ce9d282cf8